xdsl/frontend/pyast/utils/builder.py

```diff
@@ -31,8 +33,28 @@ class PyASTBuilder:
     function_ast: ast.FunctionDef
     """The AST tree for the function being built."""
 
-    desymref: bool
-    """Whether to apply the desymref flag to the built module."""
+    build_context: Context = field(default_factory=Context)
+    """The xDSL context to use when applying transformations to the built module."""
+
+    post_transforms: list[ModulePass] = field(
+        default_factory=lambda: [FrontendDesymrefyPass()]
+    )
+    """An ordered list of passes to apply to the built module."""
+
+    post_verify: bool = True
+    """Whether to verify each post processing transformation pass."""
+
+    def _apply_post_transforms(
+        self,
+        module: ModuleOp,
+        context: Context,
+        verify: bool = True,
+    ) -> None:
```

Comment:
What's the advantage of this helper over inlining the implementation? I can't think of a case of using it outsside of `build`